kroxylicious-api/src/main/java/io/kroxylicious/proxy/filter/KrpcFilterContext.java

```diff
@@ -107,5 +107,10 @@ public interface KrpcFilterContext {
      */
     ResponseFilterResultBuilder responseFilterResultBuilder();
 
+    /**
+     * Allows the filter to identify which cluster it is processing a request for
+     * @return virtual cluster name
+     */
+    String getVirtualClusterName();
```

Comment:
I'm still not 100% sold on getting clusterName via the context. Its certainly easier to implement than wiring it into the FilterContributor API but I do wonder if that is the right long term home. So my hesitation with adding it here is the churn in the public API, would we need to go through a deprecation cycle to remove it once it is available via the FilterContributor?

Reply:
Long term, I don't think it should be the virtual cluster name.  This was a pragmatic step to allow me to record a demo of the feature with the idea of publishing it soon.

I think there needs to be some kind of the filter configuration available at the virtual cluster level, which would let each virtual cluster specify a `tenantKey`.   As tenant keys are used as prefixes for topic names etc, they need to obey the rules for legal kafka resources names such as https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/common/internals/Topic.java#L36.   We should support the use case where two virtual clusters decide to use the same tenant key.

We could move filter chain definitions into the virtual cluster, but this would bring duplication.  I don't think that's the right approach.  An alternative would be to let the virtual cluster express an additional configuration source (a map), which would be made available to the filter instances.

Reply:
Sorry I should have been clearer. Yes this definitely feels like a step forward for the filter. I don't personally see a problem with using the clusterName as the tenant prefix even if it does mean having to conform to some rules (which don't seem that onerous).

@robobario what are your thoughts on the public API implications?

Reply:
I wonder if it `name()` should go within a `VirtualClusterContext` interface that is available via `KrpcFilterContext#virtualCluster()` and then we could work towards also passing `VirtualClusterContext` into `Contributor#getInstance`. This could be the interface we add our shared-state methods to? Either way it doesn't sound too onerous to deprecate/remove this method if required.

Reply:
Lets merge as is and take the conversation about how to get a name longer term somewhere else.